Quickstarts Q3 2023 update

john-rock opened the pull request:
## What are you changing in this pull request and why?

[Asana task](https://app.asana.com/0/1200099998847559/1205319420559417/f)

This PR introduces enhancements to the Quickstart layout, providing users with a more powerful and intuitive experience. Here's what's new:

1. **Filter by Topic** 🧩
   - Users can now filter quickstarts by topics of interest.
   - Introduces the `tags` markdown property on quickstarts as an array where users can specify the desired topics for filtering.

2. **Filter by Level** 📊
   - Quickstarts can be filtered based on their complexity level.
   - The `level` property, a string, allows users to set and filter by the desired level.

3. **Search Bar** 🔍
   - A convenient search bar has been added to enable users to quickly find specific quickstarts.

Additionally, added the `recently_updated` property, a boolean which when set to true will display an "updated" label on quickstarts that have been recently modified.
[
Quickstart notion doc](https://www.notion.so/dbtlabs/Quickstart-Guides-73c57db4b37547aaab4b7d9437e19eb1) has been updated as well outlining the features above.

## Preview Link
https://docs-getdbt-com-git-quickstarts-q32023-update-dbt-labs.vercel.app/guides

## Loom Walkthrough
https://www.loom.com/share/bce7a1b7ec944e109c0f251bd8566dc3

## Next Steps
This PR adds sample data to each quickstart `tags`, `level`, and `recently_updated` properties in order to test the functionality. A seperate PR will need to be opened applying the correct information on each quickstart.

Any work involved with the [migration](https://www.notion.so/dbtlabs/Quickstarts-Q3-updates-SOW-b36903d2fffc4320b27a594017010214?pvs=4#b6e72839f3364923b1da7b4b61fb3090) part of this task can begin to be worked on.

## Must be completed before merging
- [x] `/quickstarts` directory changed to `/guides`
- [x] Quickstarts updated with correct tagging
- [x] Migration work completed

john-rock committed:
start work

john-rock committed:
get multi select working

john-rock committed:
update package lock

john-rock committed:
get search working

john-rock committed:
small performance refactors

john-rock committed:
test prod build with styles

john-rock committed:
throw in input css file

john-rock committed:
styles wip

john-rock committed:
keep working on styles

john-rock committed:
add porp to deub prod build

john-rock committed:
revert for debug

john-rock committed:
update active and focus styles

john-rock committed:
small formatting update

john-rock committed:
more style wips

john-rock committed:
update selectors

john-rock committed:
get data on individual pages

john-rock committed:
styles wip

john-rock committed:
continue styling

john-rock committed:
cleanup mobile styles

john-rock committed:
add arrows to nav buttons

john-rock committed:
style cleanup

john-rock committed:
build mobile functionality

john-rock committed:
add some checks

john-rock committed:
add search icon

john-rock committed:
update dark mode styles

john-rock committed:
add some comments

john-rock committed:
sort by alphabetically

john-rock committed:
Merge branch 'current' into quickstarts-q32023-update

john-rock committed:
update package json

john-rock committed:
Merge branch 'quickstarts-q32023-update' of https://github.com/dbt-labs/docs.getdbt.com into quickstarts-q32023-update

JKarlavige commented on website/src/components/quickstartGuideList/index.js:
```diff
@@ -1,19 +1,68 @@
+  // UseMemo to prevent re-rendering on every filter change
+  // Get tag options
+  // Populated from the tags frontmatter array
+  const tagOptions = useMemo(() => {
+    const tags = new Set();
+    quickstartData.forEach(guide =>
+      guide?.data?.tags?.forEach(tag => tags.add(tag))
+    );
+    // Sort alphabetically
+    return Array.from(tags).sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase())).map(tag => ({ value: tag, label: tag }));
+  }, [quickstartData]);
```
Would be interested in diving into this more as i'm not too familiar with `useMemo`!

JKarlavige commented on website/src/components/quickstartTOC/styles.module.css:
```diff
@@ -88,10 +90,20 @@ html[data-theme="dark"] .tocList .active span {
     margin-left: auto;
 }
 
+.buttonContainer .nextButton i {
+    font-size: .8rem;
+    margin-left: .4rem;
+}
+
```
When on step 1 of a quickstart, the `Next` button at the bottom of the page has green text with a green button when in dark mode. Once you move to step 2 and on, the button has a dark bg and is readable.
![image](https://github.com/dbt-labs/docs.getdbt.com/assets/13647110/934b0fcd-7b05-4c57-a926-b32e618625ff)

JKarlavige commented on website/src/components/selectDropdown/index.js:
```diff
@@ -0,0 +1,21 @@
+
+export const SelectDropdown = ({ options, value, onChange, isMulti, placeHolder }) => {
+  return (
+    <Select
+      isMulti={isMulti}
+      name="multi-select-dropdown"
+      options={options}
+      className={styles.selectContainer}
+      classNamePrefix={styles.select}
+      value={value}
+      onChange={onChange}
+      placeholder={placeHolder}
+      aria-label="Multi Select Dropdown"
+      unstyled
+    />
```
This component looks and works great!

JKarlavige reviewed (changes requested):
Dang this came out awesome, great work! All filtering/searching looks to function correctly. Just have one small requested change for the `Next` button on the individual quickstart pages.

john-rock committed:
update colors

john-rock committed:
change quickstart dir to guides

john-rock committed:
migration work for guides

john-rock committed:
fix hover color

john-rock committed:
update quickstart links to guides

john-rock committed:
try fixing merge conflict

john-rock committed:
update broken markdown links

john-rock committed:
update another md link

john-rock committed:
update more md links

john-rock committed:
updating more broken links

john-rock committed:
more more more

john-rock committed:
update links on the cards

john-rock committed:
Merge branch 'current' into quickstarts-q32023-update

john-rock committed:
update button color

john-rock commented:
Recent commits include updating `/quickstarts` to `/guides` per @runleonarun request in the [task](https://app.asana.com/0/0/1205319420559417/1205817625817418/f).

john-rock committed:
update redirects

john-rock committed:
add another redirect

JKarlavige commented on website/docs/quickstarts/bigquery-qs.md:
Is this bigquery file within the quickstarts safe to be deleted?

JKarlavige commented on website/docusaurus.config.js:
```diff
@@ -134,8 +134,8 @@ var siteSettings = {
             {
-              label: "Quickstarts",
-              to: "/quickstarts",
+              label: "Guides",
+              to: "/guides",
```
Right above this nav item there is another `Guides` nav item which links to `/guides/best-practices`.  Thinking we'll have to rename this existing nav item.
![image](https://github.com/dbt-labs/docs.getdbt.com/assets/13647110/748dd08a-d213-4b8a-8af4-5383ac968f61)

john-rock replied:
Ope

john-rock replied:
@runleonarun do you have a preference on what we want to rename/replace this with?

runleonarun replied:
@john-rock That should be "Best practices" for now and everything in that best-practices folder should live in website/docs/best-practices/

john-rock replied:
Got it thanks @runleonarun! That's updated now.

JKarlavige reviewed (changes requested):
Just have one requested change in the nav, as there are now two `Guides` nav items. Clicked through all guides and looks to be functioning correctly!

john-rock committed:
remove old quickstart

john-rock committed:
update styles

john-rock committed:
update best practice link in nav

JKarlavige reviewed (approved):
lgtm! :shipit:

runleonarun committed:
Merge branch 'current' into quickstarts-q32023-update

runleonarun commented:
@JKarlavige  and @john-rock I'm going to go ahead and try to fix the merge conflict. 🤞🏻

runleonarun committed:
adding guides to cool new section

runleonarun committed:
fixing best-practice links

john-rock commented:
Awesome thanks @runleonarun! With this approved you should be good to move forward with the migration process.

john-rock committed:
update colors

john-rock committed:
update styles

runleonarun committed:
Merge branch 'quickstarts-q32023-update' into quickstart-content-updates

runleonarun committed:
moving more guides

runleonarun committed:
Merge branch 'quickstarts-q32023-update' of github.com:dbt-labs/docs.getdbt.com into quickstarts-q32023-update

runleonarun committed:
Merge branch 'current' into quickstarts-q32023-update

runleonarun committed:
Merge branch 'quickstarts-q32023-update' into quickstart-content-updates

runleonarun committed:
Merge branch 'current' into quickstarts-q32023-update

runleonarun commented:
Note that the content work is being done [over in this PR](https://github.com/dbt-labs/docs.getdbt.com/pull/4361/files)

runleonarun committed:
Merge branch 'quickstarts-q32023-update' into quickstart-content-updates

runleonarun committed:
Merge branch 'current' into quickstarts-q32023-update

runleonarun committed:
Merge branch 'quickstarts-q32023-update' into quickstart-content-updates

runleonarun commented:
@john-rock I'm curious about the front matter `platform: 'dbt-core'` It seems redundant to the `tags: ['migration', 'dbt Core','dbt Cloud']`. Or is there a way to sort by platform? Thank you! Sorry if I missed this.

runleonarun committed:
moving a few more guides

runleonarun committed:
fixing links

runleonarun committed:
Merge branch 'current' into quickstarts-q32023-update

runleonarun committed:
Merge branch 'quickstarts-q32023-update' into quickstart-content-updates

runleonarun committed:
moving more guides

runleonarun committed:
adding feedback from @nghi-ly

runleonarun committed:
Merge branch 'current' into quickstarts-q32023-update

runleonarun committed:
Merge branch 'quickstarts-q32023-update' into quickstart-content-updates

runleonarun committed:
commenting out time_to_complete

runleonarun committed:
adding some consistency

runleonarun committed:
moving webhooks guide content

runleonarun committed:
movign more guide content

runleonarun committed:
Merge branch 'current' into quickstarts-q32023-update

runleonarun committed:
movign snowpark

runleonarun committed:
Merge branch 'quickstarts-q32023-update' into quickstart-content-updates

runleonarun committed:
move more guides

runleonarun committed:
moving and deleting adapter stuff

runleonarun committed:
adding tags

runleonarun committed:
fixing details

runleonarun committed:
fixing links

runleonarun committed:
Merge branch 'current' into quickstarts-q32023-update

runleonarun committed:
fixing merge conflict

runleonarun committed:
fixing links

runleonarun committed:
fixing more links

runleonarun committed:
fix a link o'rama

runleonarun committed:
fix a link o'rama

runleonarun committed:
fix a typo

runleonarun committed:
adding the forwarders

runleonarun committed:
Update website/docs/docs/dbt-versions/release-notes/07-June-2023/product-docs-jun.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun committed:
Update website/docs/guides/dbt-models-on-databricks.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun committed:
Update website/docs/guides/debug-schema-names.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun committed:
Update website/docs/guides/debug-schema-names.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun committed:
Update website/docs/guides/debug-schema-names.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun committed:
Update website/docs/guides/serverless-datadog.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun committed:
Update website/docs/guides/debug-schema-names.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun committed:
Apply suggestions from code review

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

nghi-ly committed:
Update website/docs/best-practices/best-practice-workflows.md

don't need the domain for links when page is within our own docs site

nghi-ly committed:
Update website/blog/2023-04-24-framework-refactor-alteryx-dbt.md

don't need the domain for links when page is within our own docs site

nghi-ly committed:
Update website/blog/2023-04-24-framework-refactor-alteryx-dbt.md

don't need the domain for links when page is within our own docs site

nghi-ly committed:
Update website/docs/best-practices/how-we-structure/5-semantic-layer-marts.md

don't need the domain for links when page is within our own docs site

nghi-ly committed:
Update website/docs/best-practices/how-we-structure/5-semantic-layer-marts.md

don't need the domain for links when page is within our own docs site

nghi-ly committed:
Update website/docs/best-practices/how-we-style/6-how-we-style-conclusion.md

don't need the domain for links when page is within our own docs site

nghi-ly committed:
Update website/docs/docs/cloud/dbt-cloud-ide/dbt-cloud-tips.md

fix link

nghi-ly committed:
Update website/docs/docs/dbt-versions/release-notes/24-Nov-2022/dbt-databricks-unity-catalog-support.md

nghi-ly committed:
Update website/docs/docs/deploy/deploy-environments.md

nghi-ly committed:
Update website/docs/docs/deploy/deploy-environments.md

nghi-ly committed:
Update website/docs/docs/environments-in-dbt.md

nghi-ly committed:
Update website/docs/faqs/Project/multiple-resource-yml-files.md

nghi-ly committed:
Update website/docs/faqs/Project/resource-yml-name.md

nghi-ly committed:
Update website/docs/faqs/Project/structure-a-project.md

nghi-ly committed:
Update website/docs/guides/debug-schema-names.md

nghi-ly committed:
Update website/docs/guides/debug-schema-names.md

nghi-ly committed:
Update website/docs/sql-reference/aggregate-functions/sql-array-agg.md

nghi-ly committed:
Update website/docs/sql-reference/aggregate-functions/sql-avg.md

nghi-ly committed:
Update website/docs/sql-reference/aggregate-functions/sql-round.md

nghi-ly committed:
Update website/docs/sql-reference/clauses/sql-limit.md

nghi-ly committed:
Update website/docs/sql-reference/clauses/sql-order-by.md

nghi-ly committed:
Update website/docs/sql-reference/joins/sql-self-join.md

nghi-ly committed:
Update website/docs/sql-reference/joins/sql-left-join.md

nghi-ly committed:
Update website/docs/sql-reference/joins/sql-inner-join.md

nghi-ly committed:
Update website/docs/guides/custom-cicd-pipelines.md

per style guide, use sentence case for titles

nghi-ly committed:
Update website/docs/docs/cloud/billing.md

nghi-ly committed:
Update website/docs/docs/dbt-versions/release-notes/07-June-2023/product-docs-jun.md

nghi-ly committed:
Fix old redirects, remove duplicate steps

nghi-ly committed:
Moving the guide content to the new format (#4361)

## What are you changing in this pull request and why?

Moving content to new guide area and best practices section. 

This is part of the quickstart v2 project described in these docs:
* The [project
outline](https://www.notion.so/dbtlabs/Quickstarts-on-docs-V2-a457f7d673f94ee5affd5865c1dd91e3?pvs=4)
* guides [inventory and migration
spreadsheet](https://docs.google.com/spreadsheets/d/1xkBkyw7lRG7LSVXMfnKQa1rjPHZpvil8iaE8DuDhK_M/edit#gid=0)
* Internal [Docs
project](https://www.notion.so/dbtlabs/Quickstarts-V2-3d614e8cab614e26a3fb7c937a47f29b?pvs=4)

## Checklist

- [x] Review the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
and [About
versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this
PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [x] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [x] Remove page from `website/sidebars.js`
- [ ] Add an entry `website/static/_redirects`
- [ ] [Ran link
testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally)
to update the links that point to the deleted page

nghi-ly committed:
Merge branch 'current' into quickstarts-q32023-update

nghi-ly commented:
helping leona usher this to production while she's OOO. [slack convo](https://dbt-labs.slack.com/archives/C02NCQ9483C/p1699582238121899?thread_ts=1699567425.027689&cid=C02NCQ9483C)
